### PR TITLE
Removed outside iors from bssrdfs

### LINKED ISSUE #1044: Remove outside IOR params from BSSRDFs
Now that appleseed does IOR tracking, they can be automatically computed.

### PR DESCRIPTION
**Pull request by est77**
Fixes #1044

### CODE REVIEW

File: src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp [hunk 1]
```diff
@@ -87,13 +87,20 @@ namespace
             return Model;
         }
 
-        virtual void prepare_inputs(void* data) const APPLESEED_OVERRIDE
+        virtual void prepare_inputs(
+            const ShadingPoint& shading_point,
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp]: Wondering if this shouldn't be called `outgoing_point` to make it clear which point it is...

File: src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp [hunk 2]
```diff
@@ -87,13 +87,20 @@ namespace
...
+            void*               data) const APPLESEED_OVERRIDE
         {
             DipoleBSSRDFInputValues* values =
                 reinterpret_cast<DipoleBSSRDFInputValues*>(data);
 
             // Precompute the relative index of refraction.
-            values->m_eta = values->m_outside_ior / values->m_inside_ior;
+            const double outside_ior =
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp]: Are you sure about the logic here?

For comparison, here is the logic in the specular BTDF:

    if (shading_point.is_entering())
    {
        values->m_from_ior =
            shading_point.get_ray().get_current_ior();
        values->m_to_ior = values->m_ior;
    }
    else
    {
        values->m_to_ior =
            shading_point.get_ray().get_previous_ior();
        values->m_from_ior = values->m_ior;
    }

File: src/appleseed/renderer/modeling/bssrdf/normalizeddiffusionbssrdf.cpp [hunk 3]
```diff
@@ -363,17 +369,7 @@ DictionaryArray NormalizedDiffusionBSSRDFFactory::get_input_metadata() const
...
-            .insert("min_value", "1.0")
-            .insert("max_value", "2.5")
-            .insert("use", "required")
-            .insert("default", "1.0"));
-
-    metadata.push_back(
-        Dictionary()
-            .insert("name", "inside_ior")
+            .insert("name", "ior")
             .insert("label", "Inside Index of Refraction")
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/normalizeddiffusionbssrdf.cpp]: Update label.

File: src/appleseed/renderer/modeling/bssrdf/gaussianbssrdf.cpp [hunk 4]
```diff
@@ -288,17 +296,7 @@ DictionaryArray GaussianBSSRDFFactory::get_input_metadata() const
...
-            .insert("min_value", "1.0")
-            .insert("max_value", "2.5")
-            .insert("use", "required")
-            .insert("default", "1.0"));
-
-    metadata.push_back(
-        Dictionary()
-            .insert("name", "inside_ior")
+            .insert("name", "ior")
             .insert("label", "Inside Index of Refraction")
```
Comment (dictoon) [inline src/appleseed/renderer/modeling/bssrdf/gaussianbssrdf.cpp]: Update label.

**Comment by dictoon** [comment]
Nice! A welcome change. A few things to fix/review and it will be good to merge.

File: src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp (hunk 1 above)
Comment (est77) [inline src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp]: prepare inputs is independent of incoming or outgoing points.
There's only one shading point, the point at which you are going to 
evaluate the bssrdf.

File: src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp (hunk 2 above)
Comment (est77) [inline src/appleseed/renderer/modeling/bssrdf/betterdipolebssrdf.cpp]: I checked with a small sample scene and the debugger. It seems it does the right thing.

**Comment by dictoon** [comment]
Looking good, thanks!

**Pull request merged**